### ISSUE TITLE
install issue

### ISSUE THREAD

**Issue by NSLogZfRee** [leoafarias/fvm#717]
brew install fvm
==> Fetching leoafarias/fvm/fvm
==> Downloading https://storage.googleapis.com/dart-archive/channels/stable/release/3.2.6/sdk
curl: (35) Recv failure: Connection reset by peer                                           

Error: fvm: Failed to download resource "fvm--dart-sdk"
Download failed: https://storage.googleapis.com/dart-archive/channels/stable/release/3.2.6/sdk/dartsdk-macos-arm64-release.zip

what can i do ?  change network and retry?

**Comment by lkervran** [comment]
Also having an install issue with flutter 3.16.17 using `dart pub global activate fvm`

fvm flutter --version
```sh
Flutter 3.16.7 • channel stable • https://github.com/flutter/flutter.git
Framework • revision ef1af02aea (3 months ago) • 2024-01-11 15:19:26 -0600
Engine • revision 4a585b7929
Tools • Dart 3.2.4 • DevTools 2.28.5
```

fvm --version
```sh
3.1.3
```
fvm dart pub global activate fvm
```sh
Package fvm is currently active at version 3.1.3.
  dart_console 1.2.0 (4.0.2 available)
< ffi 2.1.0 (was 2.1.2) (2.1.2 available)
< http 1.2.0 (was 1.2.1) (1.2.1 available)
  intl 0.18.1 (0.19.0 available)
< web 0.4.2 (was 0.5.1) (0.5.1 available)
< win32 5.2.0 (was 5.4.0) (5.4.0 available)
Building package executables...
Built fvm:compile.
Failed to build fvm:main:
../.pub-cache/hosted/pub.dev/mason_logger-0.2.13/lib/src/ffi/windows_terminal.dart:9:33: Error: The getter 'STD_HANDLE' isn't defined for the class 'WindowsTerminal'.
 - 'WindowsTerminal' is from 'package:mason_logger/src/ffi/windows_terminal.dart' ('../.pub-cache/hosted/pub.dev/mason_logger-0.2.13/lib/src/ffi/windows_terminal.dart').
Try correcting the name to the name of an existing getter, or defining a getter or field named 'STD_HANDLE'.
    outputHandle = GetStdHandle(STD_HANDLE.STD_OUTPUT_HANDLE);
                                ^^^^^^^^^^
../.pub-cache/hosted/pub.dev/mason_logger-0.2.13/lib/src/ffi/windows_terminal.dart:10:32: Error: The getter 'STD_HANDLE' isn't defined for the class 'WindowsTerminal'.
 - 'WindowsTerminal' is from 'package:mason_logger/src/ffi/windows_terminal.dart' ('../.pub-cache/hosted/pub.dev/mason_logger-0.2.13/lib/src/ffi/windows_terminal.dart').
Try correcting the name to the name of an existing getter, or defining a getter or field named 'STD_HANDLE'.
    inputHandle = GetStdHandle(STD_HANDLE.STD_INPUT_HANDLE);
                               ^^^^^^^^^^
../.pub-cache/hosted/pub.dev/mason_logger-0.2.13/lib/src/ffi/windows_terminal.dart:18:22: Error: Undefined name 'CONSOLE_MODE'.
    const dwMode = (~CONSOLE_MODE.ENABLE_ECHO_INPUT) &
                     ^^^^^^^^^^^^
../.pub-cache/hosted/pub.dev/mason_logger-0.2.13/lib/src/ffi/windows_terminal.dart:19:11: Error: Undefined name 'CONSOLE_MODE'.
        (~CONSOLE_MODE.ENABLE_PROCESSED_INPUT) &
          ^^^^^^^^^^^^
../.pub-cache/hosted/pub.dev/mason_logger-0.2.13/lib/src/ffi/windows_terminal.dart:20:11: Error: Undefined name 'CONSOLE_MODE'.
        (~CONSOLE_MODE.ENABLE_LINE_INPUT) &
          ^^^^^^^^^^^^
../.pub-cache/hosted/pub.dev/mason_logger-0.2.13/lib/src/ffi/windows_terminal.dart:21:11: Error: Undefined name 'CONSOLE_MODE'.
        (~CONSOLE_MODE.ENABLE_WINDOW_INPUT);
          ^^^^^^^^^^^^
../.pub-cache/hosted/pub.dev/mason_logger-0.2.13/lib/src/ffi/windows_terminal.dart:27:20: Error: Undefined name 'CONSOLE_MODE'.
    const dwMode = CONSOLE_MODE.ENABLE_ECHO_INPUT |
                   ^^^^^^^^^^^^
../.pub-cache/hosted/pub.dev/mason_logger-0.2.13/lib/src/ffi/windows_terminal.dart:28:9: Error: Undefined name 'CONSOLE_MODE'.
        CONSOLE_MODE.ENABLE_EXTENDED_FLAGS |
        ^^^^^^^^^^^^
../.pub-cache/hosted/pub.dev/mason_logger-0.2.13/lib/src/ffi/windows_terminal.dart:29:9: Error: Undefined name 'CONSOLE_MODE'.
        CONSOLE_MODE.ENABLE_INSERT_MODE |
        ^^^^^^^^^^^^
../.pub-cache/hosted/pub.dev/mason_logger-0.2.13/lib/src/ffi/windows_terminal.dart:30:9: Error: Undefined name 'CONSOLE_MODE'.
        CONSOLE_MODE.ENABLE_LINE_INPUT |
        ^^^^^^^^^^^^
../.pub-cache/hosted/pub.dev/mason_logger-0.2.13/lib/src/ffi/windows_terminal.dart:31:9: Error: Undefined name 'CONSOLE_MODE'.
        CONSOLE_MODE.ENABLE_MOUSE_INPUT |
        ^^^^^^^^^^^^
../.pub-cache/hosted/pub.dev/mason_logger-0.2.13/lib/src/ffi/windows_terminal.dart:32:9: Error: Undefined name 'CONSOLE_MODE'.
        CONSOLE_MODE.ENABLE_PROCESSED_INPUT |
        ^^^^^^^^^^^^
../.pub-cache/hosted/pub.dev/mason_logger-0.2.13/lib/src/ffi/windows_terminal.dart:33:9: Error: Undefined name 'CONSOLE_MODE'.
        CONSOLE_MODE.ENABLE_QUICK_EDIT_MODE |
        ^^^^^^^^^^^^
../.pub-cache/hosted/pub.dev/mason_logger-0.2.13/lib/src/ffi/windows_terminal.dart:34:9: Error: Undefined name 'CONSOLE_MODE'.
        CONSOLE_MODE.ENABLE_VIRTUAL_TERMINAL_INPUT;
        ^^^^^^^^^^^^
```

**Comment by renefloor** [comment]
Installing fvm always tries to use the latest dependencies. It's depending on mason_logger and that's somehow breaking the install.
The change is made here: https://github.com/felangel/mason/pull/1286/files

I don't know of any way to pin the version of `mason_logger` to version 0.2.12 during the install, as there is no way to use a pubspec.yaml override.

**Comment by felangel** [comment]
This should be resolved now (in mason_logger 0.2.14). Let me know if you're still encountering any issues and apologies for the inconvenience!

**Comment by NSLogZfRee** [comment]
Still encountering same error. 2024-4-21 9:50. MacBook Pro.

**Comment by NSLogZfRee** [comment]
> This should be resolved now (in mason_logger 0.2.14). Let me know if you're still encountering any issues and apologies for the inconvenience!


Still encountering same error. 2024-4-21 9:50. MacBook Pro.

**Comment by felangel** [comment]
> > This should be resolved now (in mason_logger 0.2.14). Let me know if you're still encountering any issues and apologies for the inconvenience!
> 
> Still encountering same error. 2024-4-21 9:50. MacBook Pro.

Can you provide the output of `flutter --doctor` as well as the exact command you're running and the output you're seeing? Thanks!

**Comment by leoafarias** [comment]
@felangel which version should I pin for mason logger to avoid the issue? I can release an update with it

**Comment by felangel** [comment]
> @felangel which version should I pin for mason logger to avoid the issue? I can release an update with it

0.2.14 includes the fix (reverting the win32 change) and 0.2.15 includes the win32 change but requires Dart 3.3.0. Looks like fvm allows Dart >=2.17.0 so 0.2.14 of mason_logger should be the one that pub resolves and downloads. I don't think pinning a version should be necessary unless you really want to. I'd love to better understand any issues users are having 🤷
